### DISCUSSION TITLE
Advanced Device Interface filter

### DISCUSSION THREAD

**Discussion by chubblyButs** [netbox-community/netbox#5629] [Ideas]
Can the device interface filter be made to filter on other fields instead of just the "name" field?
For example filter on "enabled" ports, "Connected" ports.

**Comment by candlerb** [comment]
This is what I see in v2.10.3 at the `/dcim/interfaces/` page:

![image](https://user-images.githubusercontent.com/44789/104620131-90360d80-5686-11eb-9bde-592855d06a02.png)

There are multiple fields you can filter on.  "Enabled" is already supported as a filter.  "Management only (OOB)" was [recently added](https://github.com/netbox-community/netbox/issues/5570).

Filtering on "Connected" would be new a feature request.

**Reply by chubblyButs** [reply]
That is the wrong context, when viewing the device/X/interface tab, there is only a name filter which means I have to scroll through dozens of interface to view which connected to devices. 
![image](https://user-images.githubusercontent.com/76495175/104627360-ef832600-5664-11eb-9a90-ca44cb174897.png)

Scrolling further down long way down, I get to what I want to view on this page which are enabled and connected ports.

![image](https://user-images.githubusercontent.com/76495175/104627570-307b3a80-5665-11eb-826f-0f486c3967fa.png)

**Reply by chubblyButs** [reply]
If there was just a checkbox for enabled or even better for connected ports, that would increase the efficiency of this view. Scrolling past dozens of disabled/unconnected interfaces to view what's actually connected does not help my carpal tunnel. :)

**Reply by lastwednesday** [reply]
This is something that I was just looking for as well. Many of the devices we manage have several hundred interfaces (eg SAN directors), so only being able to filter by Name itself is pretty limiting and tedious. Being able to check by eg label, description, status, &c would be very valuable.
